packages/button/test/button.test.ts

```diff
@@ -423,6 +423,45 @@ describe('Button', () => {
             expect(el.getAttribute('aria-label')).to.equal('clickable');
         });
 
+        it('updates aria-label when label changes', async () => {
+            const el = await fixture<Button>(html`
+                <sp-button label="Initial Label">Button</sp-button>
+            `);
+
+            await elementUpdated(el);
+            expect(el.getAttribute('aria-label')).to.equal('Initial Label');
+
+            // Change the label
+            el.label = 'New Label';
+            await elementUpdated(el);
+
+            // The aria-label should also update
+            expect(el.getAttribute('aria-label')).to.equal('New Label');
+        });
+
+        it('preserves aria-label when slot content changes', async () => {
+            const el = await fixture<Button>(html`
+                <sp-button label="Test Label">Initial Content</sp-button>
+            `);
+
+            await elementUpdated(el);
+            expect(el.getAttribute('aria-label')).to.equal('Test Label');
+
+            // Change the slot content
+            el.textContent = 'Updated Content';
```

Comment:
```suggestion
            el.textContent = 'Updated content';
```

Same minor nit here; not ultra important since it's a test case but just a good habit to be in.